### PR TITLE
feat: Product Switch with Drag and Drop.

### PR DESCRIPTION
**Pull request by JKMarkowski**
#### Please provide a link to the associated issue.
fixes: https://github.com/SAP/fundamental-ngx/issues/1438
#### Please provide a brief summary of this pull request.
There are new `product switch` component and `drag and drop` directives. Right now usage of `drag and drop` is limited to `ProductSwitch`, but in future we can make it standalone directive. 
#### If this is a new feature, have you updated the documentation?
There is 1 new documentation page.

- [x] `README.md` - **Nothing to update there.**
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes) - **Updated 0.12.0 breaking changes**
- [x] Documentation Examples - **Added new page**

### CODE REVIEW

File: libs/core/src/lib/product-switch/product-switch-body/product-switch-body.component.scss [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+            content: '';
+            position: absolute;
+            left: -3px;
+            bottom: -7px;
+            border-radius: 50%;
+            width: 0.5rem;
+            height: 0.5rem;
+            border: 0.125rem solid $product-switch-dnd-line-color;
+            background-color: transparent;
+        }
```
Comment (InnaAtanasova) [inline libs/core/src/lib/product-switch/product-switch-body/product-switch-body.component.scss]: You don't need this part. We have the "circle" only on the top. 
Screen Shot 2019-10-21 at 4 36 43 PM

File: libs/core/src/lib/product-switch/product-switch-body/product-switch-body.component.scss [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        width: 100%;
+        height: 2px;
+
+        &.after {
+            bottom: 0;
+        }
+
+        &.before {
+            top: 0;
+        }
```
Comment (InnaAtanasova) [inline libs/core/src/lib/product-switch/product-switch-body/product-switch-body.component.scss]: But here you need the "circle" on the left 
Screen Shot 2019-10-21 at 4 38 04 PM

**Comment by droshev** [comment]
RTL mobile looks broken
Screen Shot 2019-10-21 at 7 36 59 PM

File: apps/docs/src/app/core/component-docs/product-switch/examples/product-switch-small-example.component.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        },
+        {
+            title: 'Strategic Procurement',
+            icon: 'cart-3'
+        }
+    ];
+
+    productChangeHandle(products: ProductSwitchItem[]): void {
+        this.list = products;
+        console.log(products);
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply apps/docs/src/app/core/component-docs/product-switch/examples/product-switch-small-example.component.ts]: It is in example. I wanted to show users, the new structure of the array.

File: apps/docs/src/app/core/component-docs/product-switch/examples/product-switch-list/product-switch-list.component.ts [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+        },
+        {
+            title: 'S/4HANA',
+            icon: 'batch-payments'
+        },
+    ];
+
+    productChangeHandle(products: ProductSwitchItem[]): void {
+        this.list = products;
+        console.log(products);
```
[the remark this answers is not in the document]
Comment (JKMarkowski) [reply apps/docs/src/app/core/component-docs/product-switch/examples/product-switch-list/product-switch-list.component.ts]: It is in example. I wanted to show users, the new structure of the array.

**Comment by droshev** [comment]
In LTR mode we have icons and text aligned on the left, but on RTL the icons are aligned on the left but
the text is still aligned on the right side:
Screen Shot 2019-10-22 at 7 23 22 AM

**Comment by InnaAtanasova** [comment]
> In LTR mode we have icons and text aligned on the left, but on RTL the icons are aligned on the left but
> the text is still aligned on the right side:
> Screen Shot 2019-10-22 at 7 23 22 AM

This is broken from "my side". I will fix it in Fundamental-styles. For the current version of fund-styles we can use the fix Jedrzej added. Thanks @JKMarkowski for fixing it

**Pull request merged**